parser/xml/xml.go

```diff
@@ -0,0 +1,24 @@
+package xml
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	x "github.com/basgys/goxml2json"
```

Comment:
Why not create an implementation using encoding/xml. I think that would be better then importing a small library. Any specific reason why this choice is made?

Reply:
@Blokje5  Afaik for normal `encoding/xml` use, we have to create structs for given `xml` and have to assign them to these. For example:
```go
var s Some
x.Unmarshal(p, &s) // Some is not dynamic
```
But, if you have a look at [goxml2json](https://github.com/basgys/goxml2json/blob/996d9fc8d3135d0ac85060984a7bef857e7db738/decoder.go#L67) , You can create dynamic `json` structs for given `xml`. You don't need any `schema` nor `structs`.
Also, they give us a good abstraction otherwise we have to write bunch of `xml` rules inside the `Unmarshall` function. 

If you have a good example of parsing `xml` to `json` with internal `encoding/xml`, we can go ahead with it.

Reply:
@boranx A I guess you are right. I thought there would be similar support like encoding/json for unmarshalling into a `map[string]interface{}` but apparently that is not possible.